apps/android/app/src/main/java/chat/simplex/app/views/chat/item/CIVoiceView.kt

```diff
@@ -0,0 +1,344 @@
+      val play = play@{
+        audioPlaying.value = AudioPlayer.start(getAppFilePath(SimplexApp.context, file.filePath ?: return@play), audioInfo.value.progressMs) {
+          audioPlaying.value = false
+        }
+      }
+      val pause = {
+        audioInfo.value = ProgressAndDuration(AudioPlayer.pause(), audioInfo.value.durationMs)
+        audioPlaying.value = false
+      }
+      val filePath = remember(file.filePath, file.fileStatus) { getLoadedFilePath(context, file) }
+      MiniAudioPlayer(filePath, audioPlaying, audioInfo)
+
+      val time = if (audioPlaying.value) audioInfo.value.progressMs else audioInfo.value.durationMs
+      val minWidth = with(LocalDensity.current) { 57.sp.toDp() }
+      val text = String.format("%02d:%02d", time / 1000 / 60, time / 1000 % 60)
+      if (hasText) {
+        fileIndicator(file, audioPlaying.value, sent, hasText, audioInfo, receiveFile, play, pause)
+        Text(
+          text,
+          Modifier
+            .padding(start = 12.dp)
+            .widthIn(min = minWidth),
+          color = HighOrLowlight,
+          fontSize = 16.sp,
+          textAlign = TextAlign.Start,
+          maxLines = 1
+        )
+      } else {
+        if (sent) {
+          Row {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+              Spacer(Modifier.height(56.dp))
+              Text(
+                text,
+                Modifier
+                  .widthIn(min = minWidth)
+                  .padding(end = 12.dp),
+                color = HighOrLowlight,
+                fontSize = 16.sp,
+                maxLines = 1
+              )
+            }
+            Column {
+              fileIndicator(file, audioPlaying.value, sent, hasText, audioInfo, receiveFile, play, pause)
+              Box(Modifier.align(Alignment.CenterHorizontally).padding(top = 6.dp)) {
+                CIMetaView(ci, metaColor)
+              }
+            }
+          }
+        } else {
+          Row {
+            Column {
+              fileIndicator(file, audioPlaying.value, sent, hasText, audioInfo, receiveFile, play, pause)
+              Box(Modifier.align(Alignment.CenterHorizontally).padding(top = 6.dp)) {
+                CIMetaView(ci, metaColor)
+              }
+            }
+            Row(verticalAlignment = Alignment.CenterVertically) {
+              Text(
+                text,
+                Modifier
+                  .widthIn(min = minWidth)
+                  .padding(start = 12.dp),
+                color = HighOrLowlight,
+                fontSize = 16.sp,
+                maxLines = 1
+              )
+              Spacer(Modifier.height(56.dp))
+            }
+          }
+        }
+      }
+    } else {
+      fileIndicator(null, false, sent, hasText, null, receiveFile, {}, {})
+      val metaReserve = if (edited)
+        "                     "
+      else
+        "                 "
+      Text(metaReserve)
+    }
+  }
+}
+
+@Composable
+private fun fileIcon(
```

Comment:
linter complains here "Composable functions that return Unit should start with an uppercase letter"
```suggestion
private fun FileIcon(
```